Packs/CohesityHelios/Integrations/CohesityHelios/CohesityHelios.py

```diff
@@ -0,0 +1,485 @@
+        :param Client:  cohesity helios client to use.
+
+        :type args: ``Dict[str, Any]``
+
+        :return: success message of the restore operation.
+        :rtype: ``str``
+    """
+    # Filter ransomware alert for given object name.
+    alert_id = ''
+    restore_properties = {}
+    object_name = args.get('object_name')
+    demisto.debug("Performing restore operation for object {name}".format(name=object_name))
+
+    resp = client.get_ransomware_alerts()
+    for alert in resp:
+        if alert['severity'] == 'kCritical' and alert['alertState'] == 'kOpen':
+            restore_properties = _get_property_dict(alert['propertyList'])
+            if restore_properties.get('object', "") == object_name:
+                alert_id = alert['id']
+                break
+
+    if alert_id == '':
+        raise ValueError('No anomalous object found by given name')
+
+    # Prepare restore vm properties.
+    request_payload = {
+        "name": "Cortex_XSOAR_triggered_restore_task_" + restore_properties["object"],
+        "type": "kRecoverVMs",
+        "vmwareParameters": {
+            "poweredOn": True,
+            "prefix": "Recover-",
+            "suffix": "-VM-" + str(get_current_millis())
+        },
+        "objects": [
+            {
+                "jobId": int(restore_properties["jobId"]),
+                "jobRunId": int(restore_properties["jobInstanceId"]),
+                "startedTimeUsecs": int(restore_properties["jobStartTimeUsecs"]),
+                "sourceName": restore_properties["object"],
+                "protectionSourceId": int(restore_properties["entityId"])
+            }
+        ]
+    }
+    cluster_id = restore_properties['cid']
+
+    # Post restore request to helios
+    resp = client.restore_vm_object(cluster_id, request_payload)
+
+    # Resolve ransomware alert.
+    client.resolve_ransomware_alert_by_id(alert_id)
+
+    return "Restored object {name}".format(name=object_name)
+
+
+def fetch_incidents_command(client: Client):
+    """ Fetches incidents since last run or past 7 days in case of first run
+        and sends them to Cortex XSOAR.
+
+        :type client: ``Client``
+        :param Client:  cohesity helios client to use
+    """
+    # Get last run details.
+    last_run = demisto.getLastRun()
+
+    # Compute start and end time to fetch for incidents.
+    start_time_millis = int(last_run.get('start_time')) if (
+        last_run and 'start_time' in last_run) else get_millis_from_date_time(
+            datetime.now() - timedelta(days=7))
+    end_time_millis = get_current_millis()
+
+    # Fetch all new incidents.
+    max_fetch = demisto.params().get('max_fetch')
+    max_fetch = int(demisto.params().get('max_fetch')) if (
+        max_fetch and max_fetch.isdigit()) else MAX_FETCH_DEFAULT
+
+    ransomware_resp = client.get_ransomware_alerts(
+        start_time_millis=start_time_millis,
+        end_time_millis=end_time_millis,
+        max_fetch=max_fetch)
```

Comment:
I am not sure if you don't have an issue with missing events here.
In case fetch incidents should return a max of 20 events per round, it should not miss events: the next round should poll for left incidents. This is to handle temporary spikes of more than 20 incidents per minute.
For example, if your limit is 20 and you actually have 25 events in the system, how would you know to retrieve those 5 left events in the next run? Please think of it, thanks.

Reply:
Our customers get at most a handful of alerts every day. So it might not be needed.  We can put the >max_fetch alerts in a local queue and add them next time fetch_incidents is called. Is that a recommended approach? We can discuss during the demo meeting.

Reply:
The recommended approach is to save the date of the last alter you have fetched. After that, in the upcoming fetch, you should ask the API to return the alerts from the saved last alert time.